src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -1128,6 +1140,345 @@ subroutine iceshelf_melt_ismip6(domain, err)
+               .and. bedTopography(iCell) < seaLevel ) then
+                
+                groundedMarineMarginMask(iCell) = 1
+                waterDepth = seaLevel - bedTopography(iCell)
+                          
+                submergedFaceArea = waterDepth * oceanEdgeLength
+                
+                if (config_uniform_face_melt_rate == 0.0) then                
+                ! Calculate ice front melt rate at each cell
+                   faceMeltSpeed(iCell) = (aSubglacial * waterDepth * & ! m s^-1
+                         (ismip6Runoff(iCell) / rhow * secPerDay)**alphaSubglacial &
+                         + B) * (TFocean(iCell) + addTFocean)**betaTF / secPerDay
+                else
+                   faceMeltSpeed(iCell) = config_uniform_face_melt_rate
+                endif
+                !calvingVelocity(iCell) = calvingVelocity(iCell) + iceFrontMeltRate ! m s^-1
+                faceMeltRateApplied(iCell) = faceMeltSpeed(iCell) * submergedFaceArea ! m^3 s^-1
+            else 
+                groundedMarineMarginMask(iCell) = 0
+                faceMeltRateApplied(iCell) = 0.0_RKIND     
+                faceMeltSpeed(iCell) = 0.0_RKIND
+            endif           
+         enddo
+
+         ! Distribute melt among neighbors
+         call distribute_face_melt_flux(meshPool, geometryPool, scratchPool, groundedMarineMarginMask, err_tmp)
+         err = ior(err, err_tmp)
+
+         ! Apply facemelt: open the Ark
+         thickness(:) = thickness(:) - meltingThickness(:)
+
+         ! update mask
+         call li_calculate_mask(meshPool, velocityPool, geometryPool, err_tmp)
+         err = ior(err, err_tmp)
+         block => block % next
+      enddo    ! associated(block)
+
+    end subroutine grounded_face_melt_ismip6
+!-----------------------------------------------------------------------
+
+!-----------------------------------------------------------------------
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  !  routine distribute_face_melt_flux
+!
+!> \brief Applies a specified melt rate along a grounded marine margin
+!> \author Trevor Hillebrand, Matthew Hoffman
+!> \date   April 2020
+!> \details Applied a specific melt rate along a grounded marine margin. 
+!> This is based on distribute_calving_flux subroutine in mpas_li_calving.F
+!> and adapted for grounded ice.
+!> The melt rate (faceMeltRateApplied) is calculated in subroutine grounded_face_melt_ismip6.
+!> The melt volume needs to be distributed in three ways:
+!> 1. We need to first remove any "thin" ice in front of this cell
+!> 2. Then we remove ice from this cell
+!> 3. If there is still additional ice to be removed, we have to recursively
+!>    remove ice "inland" of this cell until the total required mass is removed
+!-----------------------------------------------------------------------
+
+   subroutine distribute_face_melt_flux(meshPool, geometryPool, scratchPool, groundedMarineMarginMask, err)
```

Comment:
I'm not bothering to review this routine, because we know it will go away in the next PR.